src/composables/permissions.ts

```diff
@@ -0,0 +1,183 @@
+import { METHODS } from '@aeternity/aepp-sdk';
+import type {
+  IAppData,
+  IPermission,
+  PermissionRegistry,
+  PopupType,
+} from '@/types';
+import {
+  IS_EXTENSION_BACKGROUND,
+  MODAL_CONFIRM_ACCOUNT_LIST,
+  MODAL_CONFIRM_CONNECT,
+  MODAL_MESSAGE_SIGN,
+  POPUP_TYPE_ACCOUNT_LIST,
+  POPUP_TYPE_CONNECT,
+  POPUP_TYPE_MESSAGE_SIGN,
+  STORAGE_KEYS,
+} from '@/constants';
+import { aettosToAe } from '@/protocols/aeternity/helpers';
+import { openPopup } from '@/background/popupHandler';
+import migratePermissionsVuexToComposable from '@/migrations/003-permissions-vuex-to-composable';
+import { useStorageRef } from './composablesHelpers';
+import { IModalProps, useModals } from './modals';
+
+interface ITransactionSignPermissionOptions {
+  amount?: number;
+  fee?: number;
+  nameFee?: number;
+}
+
+const permissions = useStorageRef<PermissionRegistry>(
+  {},
+  STORAGE_KEYS.permissions,
+  {
+    backgroundSync: true,
+    migrations: [
+      migratePermissionsVuexToComposable,
+    ],
+  },
+);
+
+const modalAndPopupTypes: Partial<Record<METHODS, { modal: string, popup: PopupType }>> = {
+  [METHODS.subscribeAddress]: {
+    modal: MODAL_CONFIRM_CONNECT,
+    popup: POPUP_TYPE_CONNECT,
+  },
+  [METHODS.address]: {
+    modal: MODAL_CONFIRM_ACCOUNT_LIST,
+    popup: POPUP_TYPE_ACCOUNT_LIST,
+  },
+  [METHODS.signMessage]: {
+    modal: MODAL_MESSAGE_SIGN,
+    popup: POPUP_TYPE_MESSAGE_SIGN,
+  },
```

Comment:
Yes, because adding `signTransaction` here would require another big refactor. It is because if you try to use `hdWallet/signTransaction` action it also fires the `confirmTxSigning` action which opens the **modal** window that asks for the permission. I will deal with this in next task. For now I left `TODO` comment there: https://github.com/aeternity/superhero-wallet/pull/2369/files#diff-37085670e4cc4d0853535bb598e3c36b31ffb49af1badbebe2018c1591d51b9cR41

Reply:
Okay, Commented.